src/file_actions.ts

```diff
@@ -2,13 +2,25 @@ import * as vscode from 'vscode';
 
 import { NEURO } from '@/constants';
 import { formatContext, getFence, getPositionContext, getVirtualCursor, getWorkspacePath, getWorkspaceUri, isBinary, isPathNeuroSafe, logOutput, normalizePath, notifyOnCaughtException, stripTailSlashes } from '@/utils';
-import { ActionData, contextNoAccess, RCEAction, actionValidationFailure, actionValidationAccept, ActionValidationResult } from '@/neuro_client_helper';
+import { ActionData, contextNoAccess, RCEAction, actionValidationFailure, actionValidationAccept, ActionValidationResult, actionValidationRetry } from '@/neuro_client_helper';
 import { CONFIG, PermissionLevel, getPermissionLevel } from '@/config';
 import { targetedFileCreatedEvent, targetedFileDeletedEvent } from '@events/files';
 import { RCECancelEvent } from '@events/utils';
 import { addActions } from './rce';
+import { updateActionStatus } from '@events/actions';
 
 const CATEGORY_FILE_ACTIONS = 'File Actions';
+const ACTION_FAIL_NOTES = {
+    noFilePath: 'Directory path left unspecified',
+    noAccess: 'Access disallowed to targeted directory',
+    noWorkspace: 'Not in a workspace',
+    alreadyExists: 'Targeted path already exists',
+    doesntExist: 'Targeted path does not exist',
+    binaryFile: 'Targeted file is a binary file',
+    targetedFile: 'Targeted path is a file',
+    targetedFolder: 'Targeted path is a folder',
+    incorrectType: 'Targeted path is not a file',
+};
```

Comment:
Use `as const` to make the object immutable.
```suggestion
} as const;
```